doc/runtime-information.rst

```diff
@@ -59,12 +59,72 @@ bundled and in the bundle folder if it is bundled::
     bundle_dir = getattr(sys, '_MEIPASS', path.abspath(path.dirname(__file__)))
     path_to_dat = path.abspath(path.join(bundle_dir, 'other-file.dat'))
 
+Or, if you'd rather use pathlib_::
+
+    from pathlib import Path
+    import sys
+
+    if getattr(sys, 'frozen', False) and hasattr(sys, '_MEIPASS'):
+        bundle_dir = Path(sys._MEIPASS)
+    else:
+        bundle_dir = Path(__file__).parent
+
+    path_to_dat = bundle_dir.absolute() / "other-file.dat"
+
 It is always best to use absolute paths, so
 ``path.abspath(path.join(bundle_dir, 'other-file.dat'))`` is preferred over
-``path.join(bundle_dir, 'other-file.dat')``. Using absolute paths means that
-you're less likely to encounter errors that can occur when using relative
-paths.
+``path.join(bundle_dir, 'other-file.dat')``. Using relative paths can lead to
+confusing errors should you attempt to find the parent of the current working
+directory. You'll get the incorrect value ``''`` instead of the expected
+``'..'``::
+
+    >>> from os import path
+    >>> path.dirname(".")
+    ''
```

Comment:
I'm confused as to why you've addeed this section. A section explaining the differences between the current working directory and the actual file location is needed (or needs improving if it exists).

Reply:
I added it because I though the original *relative paths are unsafe* was too vague. I don't want some newbie reading the original, misunderstanding the scope of their unsafe-ness and obsessively putting `os.path.realpath` in every corner of code they can find - in the same way that newbies hear about hidden imports obsessively put `import package` for every package they use at the top of their code in the hope that it fixes their unrelated bug.

Reply:
@Legorooj Can you clarify your second sentence?

Reply:
Alright, that makes sense.